viperx/helper_test.go

```diff
@@ -365,3 +365,69 @@ func TestGetDuration(t *testing.T) {
 		})
 	}
 }
+
+func TestGetStringMapConfig(t *testing.T) {
+	l := logrus.New()
+	l.ExitFunc = func(code int) {
+		panic(code)
+	}
+
+	t.Run("suite=with-config-path", func(t *testing.T) {
+		for k, tc := range []struct {
+			f      string
+			fatals bool
+		}{
+			{f: "./stub/json/.project-stub-name.json"},
+			{f: "./stub/toml/.project-stub-name.toml"},
+			{f: "./stub/yaml/.project-stub-name.yaml"},
+			{f: "./stub/yml/.project-stub-name.yml"},
+			{f: "./stub/does-not-exist/foo.yml", fatals: true},
+		} {
+			t.Run(fmt.Sprintf("case=%d/path=%s", k, tc.f), func(t *testing.T) {
+				viper.Reset()
+
+				cfgFile = tc.f
+				if tc.fatals {
+					assert.Panics(t, func() {
```

Comment:
A fatal is unfortunately not a panic - I'm not sure if we can actually catch a fatal easily here. This causes the test to fail atm :/

Reply:
Issue was that cfgFile was being set, and tests further down the chain were runnign with it in a bad state, so I set it back to empty after the test